From 4232d14fc86f5d5ad261891b86ade1055da4801a Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Mon, 18 Jul 2022 11:46:33 -0500
Subject: [PATCH 1/4] Update python scripts

- Update python shebangs to explicit python3 to correct EL8 RPM build
  errors
- Run python scripts through 2to3
---
 bin/print_osg_repos        |  8 ++++----
 share/generate_adduser     |  8 ++++----
 share/generate_gridmap     | 12 ++++++------
 share/generate_sshauthkeys | 12 ++++++------
 share/generate_whitelists  | 10 +++++-----
 share/print_oasis_vonames  |  8 ++++----
 6 files changed, 29 insertions(+), 29 deletions(-)

diff --git a/bin/print_osg_repos b/bin/print_osg_repos
index 853603b..0a36060 100755
--- a/bin/print_osg_repos
+++ b/bin/print_osg_repos
@@ -1,6 +1,6 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
-import urllib2
+import urllib.request, urllib.error, urllib.parse
 import libxml2
 import sys
 
@@ -10,7 +10,7 @@ if (len(sys.argv) == 2) and (sys.argv[1] == '-u'):
 elif len(sys.argv) != 1:
     sys.exit("Usage: print_osg_repos [-u]")
 
-response = urllib2.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
+response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
 xml = response.read()
 doc = libxml2.parseDoc(xml)
 
@@ -27,4 +27,4 @@ for urlfield in doc.xpathEval("//VO/OASIS/OASISRepoURLs/URL"):
 
 repos.sort()
 for repo in repos:
-    print repo
+    print(repo)
diff --git a/share/generate_adduser b/share/generate_adduser
index ce9a385..ad52fbd 100755
--- a/share/generate_adduser
+++ b/share/generate_adduser
@@ -1,12 +1,12 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
-import urllib2
+import urllib.request
 import libxml2
 import sys
 import re
 
 #download vosummary (for oasis enabled)
-response = urllib2.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read()
 doc = libxml2.parseDoc(xml)
 
@@ -20,4 +20,4 @@ for vo in doc.xpathEval("//VO"):
         username = "ouser."+voname
         stagedir="/stage/oasis/"+voname
 
-        print "id -u",username,"&>/dev/null || (/usr/sbin/useradd",username,"-d","/home/login/"+username,"-p","`pwgen 24 -y -n -s` && ([ -d",stagedir,"] || (mkdir",stagedir,"&& chown",username,stagedir,"&& chgrp",username,stagedir,"&& chmod 755",stagedir,")))"
+        print("id -u",username,"&>/dev/null || (/usr/sbin/useradd",username,"-d","/home/login/"+username,"-p","`pwgen 24 -y -n -s` && ([ -d",stagedir,"] || (mkdir",stagedir,"&& chown",username,stagedir,"&& chgrp",username,stagedir,"&& chmod 755",stagedir,")))")
diff --git a/share/generate_gridmap b/share/generate_gridmap
index a12bf4e..b2d26ae 100755
--- a/share/generate_gridmap
+++ b/share/generate_gridmap
@@ -1,12 +1,12 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
-import urllib2
+import urllib.request
 import libxml2
 import sys
 import re
 
 #download vosummary (for oasis enabled)
-response = urllib2.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read()
 doc = libxml2.parseDoc(xml)
 
@@ -29,7 +29,7 @@ for vo in doc.xpathEval("//VO"):
             dn = re.sub(r'[^/ ()=\.\-a-zA-Z0-9\':@]+', '', dn)
             
             if original_dn != dn:
-                print >> sys.stderr, "invalid character in dn:",original_dn
+                print("invalid character in dn:",original_dn, file=sys.stderr)
                 continue
 
             #escape
@@ -45,10 +45,10 @@ for vo in doc.xpathEval("//VO"):
 
 for dn in dns:
     usernames = dns[dn]
-    print "\""+dn+"\"",",".join(usernames)
+    print("\""+dn+"\"",",".join(usernames))
 
 if len(dns) < 10:
-    print >> sys.stderr, "dn count is too small:",len(dns)
+    print("dn count is too small:",len(dns), file=sys.stderr)
     sys.exit(2)
 
 sys.exit(0)
diff --git a/share/generate_sshauthkeys b/share/generate_sshauthkeys
index 22bea4e..a409283 100755
--- a/share/generate_sshauthkeys
+++ b/share/generate_sshauthkeys
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import sys
 import os
@@ -15,8 +15,8 @@ body = p.stdout.read()
 try:
     data = json.loads(body)
 except Exception as e:
-    print >>sys.stderr, 'Error decoding json: ' + str(e)
-    print >>sys.stderr, str(body)
+    print('Error decoding json: ' + str(e), file=sys.stderr)
+    print(str(body), file=sys.stderr)
     sys.exit(2)
 
 
@@ -48,7 +48,7 @@ for vo in data:
 
             sshdir = os.path.dirname(authkeysfile)
             if not os.path.exists(sshdir):
-                os.mkdir(sshdir, 0700)
+                os.mkdir(sshdir, 0o700)
                 os.chown(sshdir, uid, gid)
 
             with open(authkeysfile + '.new', 'w') as file:
@@ -56,10 +56,10 @@ for vo in data:
             os.chown(authkeysfile + '.new', uid, gid)
 
             if origauthkeys == '':
-                print 'Creating ' + authkeysfile + ':'
+                print('Creating ' + authkeysfile + ':')
                 sys.stdout.write(newauthkeys)
             else:
-                print 'Updating ' + authkeysfile + ':'
+                print('Updating ' + authkeysfile + ':')
                 sys.stdout.flush()
                 subprocess.call(['diff', authkeysfile, authkeysfile + '.new'])
 
diff --git a/share/generate_whitelists b/share/generate_whitelists
index 412a9bc..8651c8c 100755
--- a/share/generate_whitelists
+++ b/share/generate_whitelists
@@ -1,6 +1,6 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
-import urllib2
+import urllib.request
 import libxml2
 import sys
 import os
@@ -16,13 +16,13 @@ if len(sys.argv) > 1:
     usage()
 
 def logmsg(msg):
-    print time.asctime(time.localtime(time.time())) + ' ' + msg
+    print(time.asctime(time.localtime(time.time())) + ' ' + msg)
     sys.stdout.flush()
 
 logmsg('Starting')
 
 #download oasis vosummary
-response = urllib2.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
+response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
 xml = response.read()
 doc = libxml2.parseDoc(xml)
 
@@ -47,7 +47,7 @@ for urlfield in doc.xpathEval("//VO/OASIS/OASISRepoURLs/URL"):
         if url == savedurl:
             # no change
             continue
-        print urlfile + " changed from " + savedurl + " to " + url
+        print(urlfile + " changed from " + savedurl + " to " + url)
     # whitelist does not exist or url has changed.  add it.
     cmd = 'add_osg_repository ' + url
     logmsg('Running ' + cmd)
diff --git a/share/print_oasis_vonames b/share/print_oasis_vonames
index 7057dca..672ed51 100755
--- a/share/print_oasis_vonames
+++ b/share/print_oasis_vonames
@@ -1,11 +1,11 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
-import urllib2
+import urllib.request
 import libxml2
 import sys
 import re
 
-response = urllib2.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read()
 doc = libxml2.parseDoc(xml)
 
@@ -18,4 +18,4 @@ for vo in doc.xpathEval("//VO"):
 
 vonames.sort()
 for voname in vonames:
-    print voname
+    print(voname)
-- 
2.36.1


From 6958eb329ef46f45227b83decd52c3aac047ed60 Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Mon, 18 Jul 2022 12:06:03 -0500
Subject: [PATCH 2/4] libxml2 expects str rather than bytes

---
 bin/print_osg_repos       | 2 +-
 share/generate_adduser    | 2 +-
 share/generate_gridmap    | 2 +-
 share/generate_whitelists | 2 +-
 share/print_oasis_vonames | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/bin/print_osg_repos b/bin/print_osg_repos
index 0a36060..f3a4541 100755
--- a/bin/print_osg_repos
+++ b/bin/print_osg_repos
@@ -11,7 +11,7 @@ elif len(sys.argv) != 1:
     sys.exit("Usage: print_osg_repos [-u]")
 
 response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
-xml = response.read()
+xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
 repos = []
diff --git a/share/generate_adduser b/share/generate_adduser
index ad52fbd..2b9ecdb 100755
--- a/share/generate_adduser
+++ b/share/generate_adduser
@@ -7,7 +7,7 @@ import re
 
 #download vosummary (for oasis enabled)
 response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
-xml = response.read()
+xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
 #parse out the VO names
diff --git a/share/generate_gridmap b/share/generate_gridmap
index b2d26ae..ca231ec 100755
--- a/share/generate_gridmap
+++ b/share/generate_gridmap
@@ -7,7 +7,7 @@ import re
 
 #download vosummary (for oasis enabled)
 response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
-xml = response.read()
+xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
 dns = {}
diff --git a/share/generate_whitelists b/share/generate_whitelists
index 8651c8c..b4f68bd 100755
--- a/share/generate_whitelists
+++ b/share/generate_whitelists
@@ -23,7 +23,7 @@ logmsg('Starting')
 
 #download oasis vosummary
 response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
-xml = response.read()
+xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
 # NOTE: although this downloads whitelists for all registered repositories,
diff --git a/share/print_oasis_vonames b/share/print_oasis_vonames
index 672ed51..873e9ac 100755
--- a/share/print_oasis_vonames
+++ b/share/print_oasis_vonames
@@ -6,7 +6,7 @@ import sys
 import re
 
 response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
-xml = response.read()
+xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
 vonames = []
-- 
2.36.1


From 8b7cc2dcc3e8fa61b1768774d1fb94c07e801b80 Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Mon, 18 Jul 2022 12:09:37 -0500
Subject: [PATCH 3/4] Convert tabs to spaces

---
 share/print_oasis_vonames | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/share/print_oasis_vonames b/share/print_oasis_vonames
index 873e9ac..446714a 100755
--- a/share/print_oasis_vonames
+++ b/share/print_oasis_vonames
@@ -13,8 +13,8 @@ vonames = []
 for vo in doc.xpathEval("//VO"):
     voname = vo.xpathEval("Name")[0].content
     if vo.xpathEval("OASIS/Managers/Manager"):
-	voname = re.sub(r'[^/a-z0-9]+', '', voname.lower())
-	vonames.append(voname)
+        voname = re.sub(r'[^/a-z0-9]+', '', voname.lower())
+        vonames.append(voname)
 
 vonames.sort()
 for voname in vonames:
-- 
2.36.1


From 9d2f0b3c28231d9f2aeee984a3303acfe1c95449 Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Mon, 18 Jul 2022 12:13:38 -0500
Subject: [PATCH 4/4] Swap OSG registry URLs from my.osg to topology.osg

---
 bin/print_osg_repos       | 2 +-
 share/generate_adduser    | 2 +-
 share/generate_gridmap    | 2 +-
 share/generate_whitelists | 2 +-
 share/print_oasis_vonames | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/bin/print_osg_repos b/bin/print_osg_repos
index f3a4541..3df7c07 100755
--- a/bin/print_osg_repos
+++ b/bin/print_osg_repos
@@ -10,7 +10,7 @@ if (len(sys.argv) == 2) and (sys.argv[1] == '-u'):
 elif len(sys.argv) != 1:
     sys.exit("Usage: print_osg_repos [-u]")
 
-response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
+response = urllib.request.urlopen("https://topology.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
 xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
diff --git a/share/generate_adduser b/share/generate_adduser
index 2b9ecdb..11102bb 100755
--- a/share/generate_adduser
+++ b/share/generate_adduser
@@ -6,7 +6,7 @@ import sys
 import re
 
 #download vosummary (for oasis enabled)
-response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://topology.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
diff --git a/share/generate_gridmap b/share/generate_gridmap
index ca231ec..8509385 100755
--- a/share/generate_gridmap
+++ b/share/generate_gridmap
@@ -6,7 +6,7 @@ import sys
 import re
 
 #download vosummary (for oasis enabled)
-response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://topology.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
diff --git a/share/generate_whitelists b/share/generate_whitelists
index b4f68bd..28c219c 100755
--- a/share/generate_whitelists
+++ b/share/generate_whitelists
@@ -22,7 +22,7 @@ def logmsg(msg):
 logmsg('Starting')
 
 #download oasis vosummary
-response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
+response = urllib.request.urlopen("https://topology.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&sort_key=name")
 xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
diff --git a/share/print_oasis_vonames b/share/print_oasis_vonames
index 446714a..9c73929 100755
--- a/share/print_oasis_vonames
+++ b/share/print_oasis_vonames
@@ -5,7 +5,7 @@ import libxml2
 import sys
 import re
 
-response = urllib.request.urlopen("https://my.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
+response = urllib.request.urlopen("https://topology.opensciencegrid.org/vosummary/xml?summary_attrs_showoasis=on&all_vos=on&active=on&active_value=1&oasis=on&oasis_value=1&sort_key=name")
 xml = response.read().decode(response.headers.get_content_charset(failobj='utf-8'))
 doc = libxml2.parseDoc(xml)
 
-- 
2.36.1

